jupyterhub/tests/conftest.py

```diff
@@ -45,9 +45,7 @@ def io_loop(request):
 
     def _close():
         io_loop.clear_current()
-        if (not ioloop.IOLoop.initialized() or
-                io_loop is not ioloop.IOLoop.instance()):
-            io_loop.close(all_fds=True)
+        io_loop.close(all_fds=True)
```

Comment:
Nice. I had isolated `initialized` as a problem yesterday and tried without one of the if conditions. Should have removed both. As an FYI as we think about Tornado 5.0 (as it has quite a few changes with IOLoop), pytest-asyncio is a thing now (based on pytest-tornado). I've also played a bit with MonkeyType and pyAnnotate after having a discussion with Guido about typing/debugging async.

Reply:
I think this bit of checking was from when we had multiple loops running in threads. We don't need to do all that now that we have just one loop.

I've been thinking about whether pytest-asyncio is appropriate, too. I think we still need to explicity run the tornado eventloop, I'm not sure.

Reply:
Quite likely, we may have a set of tests that use pytest-Tornado and gradually introduce others with pytest-asyncio.